File: leetcode/344. Reverse String/main.scala
object Solution {
    def reverseString(s: Array[Char]): Unit = {
      var start = 0
      var end = s.length - 1

      while (start < end) {
        val temp = s(start)
        s(start) = s(end)
        s(end) = temp

        start += 1
        end -= 1
      }
    }
}

File: leetcode/344. Reverse String/main.rs
impl Solution {
    pub fn reverse_string(s: &mut Vec<char>) {
        let (mut start, mut end) = (0, s.len() - 1);

        while start < end {
            s.swap(start, end);
            start += 1;
            end -= 1;
        }
    }
}
